test/system/src/test/java/io/pravega/test/system/ReadWriteAndAutoScaleWithFailoverTest.java

```diff
@@ -97,58 +104,65 @@ public void setup() {
         testState = new TestState();
         testState.writersListComplete.add(0, testState.writersComplete);
         testState.writersListComplete.add(1, testState.newWritersComplete);
+
+        createScopeAndStream(scope, AUTO_SCALE_STREAM, config, controllerURIDirect);
+        log.info("Scope passed to client factory {}", scope);
+
+        clientFactory = new ClientFactoryImpl(scope, controller);
+        readerGroupManager = ReaderGroupManager.withScope(scope, controllerURIDirect);
     }
 
     @After
     public void tearDown() {
         testState.stopReadFlag.set(true);
         testState.stopWriteFlag.set(true);
+        //scale the controller and segmentStore back to 1 instance.
         controllerInstance.scaleService(1, true);
         segmentStoreInstance.scaleService(1, true);
+        //interrupt writers and readers threads if they are still running.
+        testState.writers.forEach(future -> future.cancel(true));
+        testState.readers.forEach(future -> future.cancel(true));
+        clientFactory.close();
+        readerGroupManager.close();
         executorService.shutdownNow();
         testState.eventsReadFromPravega.clear();
     }
 
 
     @Test(timeout = 12 * 60 * 1000)
```

Comment:
If we are adding a global timeout, then we can remove this one.

Reply:
yes, removed it.